InventoryManagementSystem/ProductManager.java

```diff
@@ -0,0 +1,112 @@
+package InventoryManagementSystem;
+
+import java.nio.file.ProviderNotFoundException;
+import java.util.ArrayList;
+import InventoryManagementSystem.Model.Product;
+import InventoryManagementSystem.CustomExceptions.InventoryExceptions.*;
+import InventoryManagementSystem.CustomExceptions.ProductExceptions.*;
+
+public class ProductManager {
+    private ArrayList<Product> availableProducts = new ArrayList();
+
+    public void add(Product product) throws InvalidProductStateException {
+        try {
+            product.validateData();
+            if (productExists(product.getProductId())) {
+                throw new ProductAlreadyExistsException("Product already exists");
+            } else
+                availableProducts.add(product);
+        } catch (ProductAlreadyExistsException e) {
+            Product existingProduct = availableProducts.get(availableProducts.indexOf(product));
+            existingProduct.setQuantity(existingProduct.getQuantity() + 1);
+            System.out.println(
+                    "This product already exists in the inventory. Updated the quantity of already existing product.");
+        } finally {
+            System.out.println("product add operation successful");
+        }
+    }
+
+    public void delete(Product product) throws InvalidProductStateException {
+        try {
+            product.validateData();
+            if (productExists(product.getProductId())) {
+                availableProducts.remove(product);
+            } else
```

Comment:
please add curly brackets for else block also or remove curly brackets for if also.